### ISSUE TITLE
SetHealth input doesn't lower health as expected

### ISSUE THREAD

**Issue by Luke18033** [StrataSource/Portal-2-Community-Edition#1227]
### Describe the bug

The `SetHealth` and `RemoveHealth` inputs appear to not lower health correctly in Portal 2; they only do half the damage they are supposed to, resulting in the entity's health not actually being set to the value specified, but to something higher.

If the player's health is visualized (e.g. with `player_debug_print_damage` or VScript), you can see that it seems to always deal exactly half of the damage it should; for `SetHealth` it's `(current health - target health) / 2`, while for `RemoveHealth` this is `target damage / 2`.

This doesn't happen when increasing the player's health. I only checked this with players, unsure if it also affects other entities.

### Issue Map

Any

### To Reproduce

1. Turn on `player_debug_print_damage`
2. Run `ent_fire !self SetHealth 0`
3. Player does not die. Console reports only 50 damage dealt, when it should have been 100.

### Operating System

Tested on Windows 10

**Comment by Luke18033** [comment]
This seems to actually apply to all damage, not just these inputs. Portal turrets will probably need their damage halved to account for this being fixed, otherwise most existing Portal stuff is instakill so it shouldn't change much.

**Comment by Luke18033** [comment]
To add to this, I don't think we should keep this behavior for backwards compatibility, for these inputs or for anything else. Backwards compat is good up to a point, but it's really not worth it when doing so requires keeping around bugs that are incredibly unintuitive to work with (when you tell the player to take 50 damage, they should take 50 damage, not 25). Especially for something like this, since most stuff in Portal is instant death anyway, and most of the common things that aren't (e.g. turrets) use hardcoded damage values which could be halved in code to compensate for the fix.

**Comment by Luke18033** [comment]
Turns out this is actually because of a cvar `sk_dmg_take_scale1` being defaulted to `0.5`, I guess it's a weird leftover from the hl2 difficulty system then? I don't know honestly

**Comment by lenship2** [comment]
I don't think this should be changed, because there are damage sources like turrets and bombs that do get affected if you set the skill damage multipliers to their defaults. The most notable example of this breaking backwards compat is the bombs in sp_a4_finale4, where the ending booby trapped stalemate button can actually kill the player outside of "easy mode".
